Sources/Database/Realm/Protocols/ValidatedDelete.swift

```diff
@@ -0,0 +1,13 @@
+import Foundation
+
+/// Allow an object to decide if it should be deleted from the realm.
+/// Useful in cases when something else has a reference to the object.
+/// This can elegantly be implemented by using `LinkingObject`.
+/// See: https://www.mongodb.com/docs/realm-sdks/swift/latest/Structs/LinkingObjects.html
+protocol ValidatedDelete {
+    var canBeRemovedFromRealm: Bool { get }
+}
+
+extension ValidatedDelete {
```

Comment:
should we maybe implement it as such?
`extension RLMObjectBase: ValidatedDelete {`
so that we don't have to write `(object as? ValidatedDelete)?.canBeRemovedFromRealm ?? true`

Reply:
By doing this, we would conform all Realm objects to `ValidatedDelete` and we wouldn't have an opportunity to override that behaviour in specific objects (redundant conformance)